include/openssl/e_os2.h

```diff
@@ -206,8 +206,12 @@ extern "C" {
 #  endif
 # endif
 
-# if defined(OPENSSL_SYS_UEFI) && !defined(ssize_t)
-#  define ossl_ssize_t int
+# if defined(OPENSSL_SYS_UEFI) && !defined(ossl_ssize_t)
+#  if !defined(ssize_t)
+#   define ossl_ssize_t int
+#  else
+#   define ossl_ssize_t ssize_t
+#  endif
 #  define OSSL_SSIZE_MAX INT_MAX
```

Comment:
There is inconsistency here in sense that if you rely on ssize_t, then how can you be sure that OSSL_SSIZE_MAX should be INT_MAX? But ssize_t is usually typedef-ed, so it's unlike that it would be defined. I therefore would suggest (insist really:-) just:
```
# if defined(OPENSSL_SYS_UEFI) && !defined(ossl_ssize_t)
# define ossl_ssize_t int
# define OSSL_SIZE_MAX INT_MAX
#endif
```
One can also wonder if `long int` (and LONG_MAX) would be more appropriate in the context... You know because L and P usually "belong" together as in ILP32 or LP64...